src/project/git/GitCommandGitProject.ts

```diff
@@ -294,26 +319,73 @@ export class GitCommandGitProject extends NodeFsLocalProject implements GitProje
 function clone(credentials: ProjectOperationCredentials,
                id: RemoteRepoRef,
                opts: CloneOptions,
-               directoryManager: DirectoryManager): Promise<GitProject> {
+               directoryManager: DirectoryManager,
+               secondTry: boolean = false): Promise<GitCommandGitProject> {
     return directoryManager.directoryFor(id.owner, id.repo, id.sha, opts)
         .then(cloneDirectoryInfo => {
-                switch (cloneDirectoryInfo.type) {
-                    case "parent-directory" :
-                        const repoDir = `${cloneDirectoryInfo.path}/${id.repo}`;
-                        const command = (id.sha === "master") ?
-                            `git clone --depth 1 ${id.cloneUrl(credentials)}` :
-                            `git clone ${id.cloneUrl(credentials)}; cd ${id.repo};git checkout ${id.sha}`;
-
-                        logger.info(`Cloning repo '${id.url}' to '${cloneDirectoryInfo.path}'`);
-                        return exec(command, { cwd: cloneDirectoryInfo.path })
-                            .then(_ => {
-                                logger.debug(`Clone succeeded with URL '${id.url}'`);
-                                // fs.chmodSync(repoDir, "0777");
-                                return NodeFsLocalProject.fromExistingDirectory(id, repoDir);
+            switch (cloneDirectoryInfo.type) {
+                case "empty-directory" :
```

Comment:
"parent-directory" is no more; it gives you an "empty-directory"; it set up the directory for you. 
I have forgotten why this was useful but it simplified something a lot.